### PR TITLE
Test both NFS3 and 4+ 

### PR DESCRIPTION
**Pull request by benjamreis**
Add  NFS4+ only related fixtures
Parametrize tests to run on both NFS3 and 4+ fixtures.

### CODE REVIEW

**Review by Wescoeur — APPROVED** [review]

**Review by stormi — COMMENTED** [review]
I still have doubts about parametrization being the right tool for this test, especially seeing the amount of changes and the fact that we still had to duplicate fixtures.

File: tests/storage/nfs/conftest.py [hunk 1]
```diff
@@ -1,6 +1,14 @@
 import logging
 import pytest
 
+# --- Dispatch fixture for NFS versions ----------------------------------------
+
+@pytest.fixture
+def dispatch_nfs(request):
+    yield request.getfixturevalue(request.param)
+
+# --- NFS3 fixtures ------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (benjamreis) [reply tests/storage/nfs/conftest.py:10]: It can, but in fact we want to the NFS3 case here.

**Review by stormi — APPROVED** [review]

**Comment by ydirson** [comment]
@benjamreis can you give details about how those fixtures annotated with "Make sure these fixtures are called before the parametrized one" actually work and why they are needed?

**Comment by ydirson** [comment]
Checking what happens when we remove those fixtures... it makes any of those tests fail right away when they try to access the param where they collect their info.  As if the `dispatch_nfs` mechanism does not pull the dependent fixtures?  That would hardly explain why removing the params in #229 allows to get rid of those extra fixture declarations :thinking:

**Pull request merged**